### PR TITLE
Added solution for Task 4

### PR DESCRIPTION
**Pull request by 233gme**
Fixed some bugs
Added solution for task 4

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,28 +1,102 @@
...
   }
 
   render() {
+    const tasks = this.state.cards.map((card) => {
+      return (
+        <Task
+          key={card.id}
+          task={card}
+          view={this.state.viewOnlyHandler}
+          newTextChangeHandler={this.newTextChangeHandler}
```
Comment (iba-dhetsikau) [inline src/App.jsx:86]: When passing an event handler - there is an approach to add 'on' prefix to its name, so in children component we could know which of passed props is a handler to subscribe onto.

File: src/Components/Task/index.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  );
+
+  const viewState = (
+    < div className='task task_is_green'>
+      <div className='task_title'>
+        <p>{props.task.title}</p>
+      </div>
+      <p className='task_text'>{props.task.text}</p>
+    </div >
+  );
```
Comment (iba-dhetsikau) [inline src/Components/Task/index.jsx]: Looks like you've missed a task assignment detail:

> Чекбокс самой карточки (как и стили, при (не-)активном состоянии чекбокса) должен быть независим от чекбокса “только просмотр” (т.е. при переключении чекбокса на самой карточки, стили также должны меняться).

File: src/Components/Task/index.jsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  // Stage for type of button and adding input fields
+  let [editFlag, setTaskState] = useState(true);
+  const changeText = () => {
+    setClassState(classFlag = true);
+    setTaskState(!editFlag);
+  };
+
+  // Intermediate storage facility for text
+  const newTextObj = { ...props.task };
```
Comment (iba-dhetsikau) [inline src/Components/Task/index.jsx:18]: Why don't you put this temp data storage as a separate state entry? You're using a simple variable/constant for that purpose though this role is logically delegated to state object.

**Review by iba-dhetsikau — CHANGES_REQUESTED** [review]
Please apply changes according to file comments and resend for a review.

File: src/App.jsx (hunk 1 above)
Comment (233gme) [reply src/App.jsx:86]: Fixed

File: src/Components/Task/index.jsx (hunk 2 above)
Comment (233gme) [reply src/Components/Task/index.jsx]: Fixed

File: src/Components/Task/index.jsx (hunk 3 above)
Comment (233gme) [reply src/Components/Task/index.jsx:18]: Sorry, I didn't quite understand what you mean,
Instead of an object, I need to make a state...

Comment (iba-dhetsikau) [reply src/Components/Task/index.jsx:18]: Yes, that's what I've meant. Better create a state entry for that purpose.

File: src/Components/Task/index.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          <button title='сохранить изменения' onClick={saveText}><AiOutlineCheck /></button>
+          <button title='отменить изменения' onClick={changeText}><AiOutlineClose /></button>
+        </div>
+      </div>
+      <textarea className='task_textarea' name='text' onChange={newText} defaultValue={props.task.text}></textarea>
+    </div>
+  );
+
+  if (editFlag) return normalMode;
+  else return props.onView ? normalMode : editMode;
```
Comment (iba-dhetsikau) [inline src/Components/Task/index.jsx:54]: Actually you have a bug. Here is the scenario:
1. Start editing the card.
2. Check ViewOnly checkbox.
3. Uncheck ViewOnly checkbox.
4. The card appears in `editMode`, however, should be in `normalMode`.

**Review by iba-dhetsikau — APPROVED** [review]
Please apply changes according to file comments, put them into a separate commit  - before moving to next task assignment, then add to following PR for a review. Merge approved.

**Pull request merged**